Codec/Archive/Tar/Check.hs

```diff
@@ -65,24 +65,46 @@ checkSecurity = checkEntries checkEntrySecurity
 checkEntrySecurity :: Entry -> Maybe FileNameError
 checkEntrySecurity entry = case entryContent entry of
     HardLink     link -> check (entryPath entry)
-                 `mplus` check (fromLinkTarget link)
+                 `mplus` checkLink (entryPath entry) (fromLinkTarget link)
     SymbolicLink link -> check (entryPath entry)
-                 `mplus` check (fromLinkTarget link)
+                 `mplus` checkLink (entryPath entry) (fromLinkTarget link)
     _                 -> check (entryPath entry)
 
   where
+
+    checkCommon name =
+        FilePath.Native.isAbsolute name || not (FilePath.Native.isValid name)
+
     check name
-      | FilePath.Native.isAbsolute name
+      | checkCommon name
```

Comment:
Previously when `not (FilePath.Native.isValid name)` we returned `InvalidFileName`, but now it is `AbsoluteFileName`. Why?